std.fs.Dir.readFileAllocOptions() fails when alignment parameter is given.

The following simple code fails to compile.
It seems that readAllArrayList() is always expecting a non-aligned ArrayList(u8), but readToEndAllocOptions() can pass an instance of ArrayListAligned(u8, 4) to it.
```
const std = @import("std");

pub fn main() !void
{
    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
    defer arena.deinit();
    const allocator = &arena.allocator;
    const block = try std.fs.cwd().readFileAllocOptions(
        allocator, "test.zig", 1000000, null, @alignOf(u32), null);
    std.debug.print("ptr={*}\n", .{ptr});
}
```

Error message:
```
D:\Code\testzig>zig build-exe test.zig
D:\Code\zig\build\lib\zig\std\fs\file.zig:397:41: error: expected type '*std.array_list.ArrayListAligned(u8,null)', found '*std.array_list.ArrayListAligned(u8,4)'
        self.reader().readAllArrayList(&array_list, max_bytes) catch |err| switch (err) {
                                        ^
D:\Code\zig\build\lib\zig\std\fs.zig:1521:42: note: called from here
        return file.readToEndAllocOptions(allocator, max_bytes, stat_size, alignment, optional_sentinel);
                                         ^
.\test.zig:8:56: note: called from here
    const block = try std.fs.cwd().readFileAllocOptions(
                                                       ^
.\test.zig:4:1: note: called from here
{
^
D:\Code\zig\build\lib\zig\std\fs\file.zig:397:41: note: pointer type child 'std.array_list.ArrayListAligned(u8,4)' cannot cast into pointer type child 'std.array_list.ArrayListAligned(u8,null)'
        self.reader().readAllArrayList(&array_list, max_bytes) catch |err| switch (err) {
                                        ^
D:\Code\zig\build\lib\zig\std\array_list.zig:25:12: note: std.array_list.ArrayListAligned(u8,null) declared here
    return struct {
           ^
D:\Code\zig\build\lib\zig\std\array_list.zig:25:12: note: std.array_list.ArrayListAligned(u8,4) declared here
    return struct {
           ^
.\test.zig:8:19: note: referenced here
    const block = try std.fs.cwd().readFileAllocOptions(
                  ^
D:\Code\zig\build\lib\zig\std\start.zig:272:40: note: referenced here
            const result = root.main() catch |err| {
                                       ^

D:\Code\testzig>
```

--- a/lib/std/fs/file.zig
+++ b/lib/std/fs/file.zig
@@ -394,7 +394,7 @@ pub const File = struct {
         var array_list = try std.ArrayListAligned(u8, alignment).initCapacity(allocator, initial_cap);
         defer array_list.deinit();
 
-        self.reader().readAllArrayList(&array_list, max_bytes) catch |err| switch (err) {
+        self.reader().readAllArrayListAligned(alignment, &array_list, max_bytes) catch |err| switch (err) {
             error.StreamTooLong => return error.FileTooBig,
             else => |e| return e,
         };

--- a/lib/std/io/reader.zig
+++ b/lib/std/io/reader.zig
@@ -57,6 +57,15 @@ pub fn Reader(
         /// If the number of bytes appended would exceed `max_append_size`, `error.StreamTooLong` is returned
         /// and the `std.ArrayList` has exactly `max_append_size` bytes appended.
         pub fn readAllArrayList(self: Self, array_list: *std.ArrayList(u8), max_append_size: usize) !void {
+            return self.readAllArrayListAligned(null, array_list, max_append_size);
+        }
+
+        pub fn readAllArrayListAligned(
+            self: Self,
+            comptime alignment: ?u29,
+            array_list: *std.ArrayListAligned(u8, alignment),
+            max_append_size: usize
+        ) !void {
             try array_list.ensureCapacity(math.min(max_append_size, 4096));
             const original_len = array_list.items.len;
             var start_index: usize = original_len;

